src/mrpro/utils/reshape.py

```diff
@@ -99,3 +99,62 @@ def reduce_view(x: torch.Tensor, dim: int | Sequence[int] | None = None) -> torc
         for d, (oldsize, stride) in enumerate(zip(x.size(), stride, strict=True))
     ]
     return torch.as_strided(x, newsize, stride)
+
+
+# @lru_cache
```

Comment:
Not sure if lru_cache helps here.
-- I will  have to benchmark it

The code is written, such that the parameters to the function can be used as cache keys.

Reply:
I disabled it for now, if a reviewer want to debug it, it is easier without the cache.

Reply:
(lets not resovle this conversation while the commented our line is still present )